web/src/narrow_banner.ts

```diff
@@ -70,7 +74,47 @@ function retrieve_search_query_data(): SearchData {
     return search_string_result;
 }
 
-function pick_empty_narrow_banner(): NarrowBannerData {
+export function get_direct_message_error_string(user_ids_string: string): string {
```

Comment:
I think we can name it better to make it clear this functions also checks the permission. And also would be better to place this in `compose_validate` probably.

Reply:
Renamed it to `check_dm_permissions_and_get_error_string` and moved it to `compose_validate`.